src/protocols/ssh/click.c

```diff
@@ -0,0 +1,457 @@
+    int flag = sentinel;
+
+    /* Get head */
+    while ((flag == 0 || flag == 32) && (head >= 0 && head <= term->display->width)){
+        flag = term->display->operations[row * term->display->width + head].character.value;
+        head--;
+    }
+
+    /* Reset flag */
+    flag = sentinel;
+
+    /* Get tail */
+    while ((flag == 0 || flag == 32) && (tail >= 0 && tail <= term->display->width)){
+        flag = term->display->operations[row * term->display->width + tail].character.value;
+        tail++;
+    }
+                
+    head += 2;
+    tail -= 2;
+
+    if(head == 1 && (term->display->operations[row * term->display->width + head].character.value == 0 ||
+                     term->display->operations[row * term->display->width + head].character.value == 32))
+        head = 0;
+    
+    click->select_head = head;
+    click->select_tail = tail;
+
+    return click;
+}
+
+void guac_double_click(guac_click* click){
+
+    /* Set selection variables */
+    guac_terminal* term = click->term;
+    int row = click->select_row;
+    int col = click->select_col;
+    printf("Row: %d, Col: %d\n", row, col);
+    int sentinel = term->display->operations[row * term->display->width + col].character.value;
+
+    /* Determination */
+
+    /* Words */
+
+    if ((sentinel > 64 && sentinel < 91) || (sentinel > 96 && sentinel < 123) || (sentinel > 47 && sentinel < 58) || (sentinel == 95)){
+        
+        /* get border */
+        guac_click* selection = guac_click_get_word_border(click);
+
+        /* Copy to clipboard */
+        guac_click_select_word(selection);
+
+        /* Draw selection */
+        guac_click_draw_select(selection);
+
+        return;
+
+    }
+
+    /* Marks */
+
+    else if ((sentinel > 32 && sentinel < 48) || (sentinel > 57 && sentinel < 65) || (sentinel > 90 && sentinel < 95) || (sentinel > 122 && sentinel < 127) ||(sentinel == 96)){
+
+        /* Copy to clipboard */
+        guac_click_select_mark(click);
+
+        /* Draw selection */
+        click->select_head = click->select_col;
+        click->select_tail = click->select_col;
+
+        guac_click_draw_select(click);
+
+        return;
+
+    }
+
+    /* Blank */
+
+    else if (sentinel == 0 || sentinel == 32){
+
+        /* Get blank border */
+        guac_click* selection = guac_click_get_blank_border(click);
+
+        /* Blank position */
+        int head = selection->select_head;
+        int tail = selection->select_tail;
+        int width = selection->term->display->width;
+
+        if (head == 0 || tail < width - 1){
+            
+            /* get border */
+            guac_click* selection = guac_click_get_blank_border(click);
+
+            /* Copy to clipboard */
+            guac_click_select_blank(selection);
+
+            /* Draw selection */
+            guac_click_draw_select(selection);
+
+            return;
+        }
+
+        else if (head > 0 && tail == width - 1){
```

Comment:
Please watch out for the spacing - there should be a space between the parenthesis and the opening brace. Same elsewhere between `if`/`while` and the opening paren.